src/main/java/net/minecraftforge/fml/common/discovery/TweakDiscoverer.java

```diff
@@ -0,0 +1,73 @@
+package net.minecraftforge.fml.common.discovery;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+
+import java.io.InputStream;
+import java.util.List;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+import java.util.zip.ZipEntry;
+
+import net.minecraftforge.fml.common.FMLLog;
+import net.minecraftforge.fml.common.ModContainer;
+import net.minecraftforge.fml.common.TweakModContainer;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.logging.log4j.Level;
+
+public class TweakDiscoverer implements ITypeDiscoverer
+{
+
+    @Override
+    public List<ModContainer> discover(ModCandidate candidate, ASMDataTable table)
+    {
+        List<ModContainer> containers = Lists.newArrayList();
+
+        JarFile jar = null;
+        try
+        {
+            jar = new JarFile(candidate.getModContainer());
+            Attributes manifest = jar.getManifest().getMainAttributes();
+            String path = manifest.getValue("TweakMetaFile");
```

Comment:
@kashike I agree with that, but there is not a good place to put the constant. I think we should leave as is, since Forge also doesn't have a constant for "mcmod.info" which is being used in `DirectoryDiscoverer`, `JarDiscoverer` and some other parts of the code.